packages/cli/src/config/extension.ts

```diff
@@ -253,6 +253,24 @@ export function loadExtension(
       )
       .filter((contextFilePath) => fs.existsSync(contextFilePath));
 
+    // IDs are created by hashing parts of the extension metadata in order to
+    // obfuscate any potentially sensitive information such as private
+    // git urls or project names.
+    const hash = createHash('sha256');
+    switch (installMetadata?.type) {
+      case 'git':
+      case 'github-release':
+        // For git and github-release installs, identify the extension based on the
+        // source it was installed from. This allows us to distinguish between
+        // extensions that happen to have the same name but are actually different.
+        hash.update(installMetadata.source);
```

Comment:
Do we need to do any standardization/clean-up of this? (e.g., will it be 'https://github.com/org/repo' in one place and 'https://github.com/org/repo/' in another?)

Reply:
Yeah it does look like we record the source exactly as the user provided... so installing via the ssh URI is also going to be different.